src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -873,10 +874,32 @@ protected virtual void Dispose(bool disposing)
 
         public void Dispose()
         {
+            StopWorkerProcessAsync();
             _disposing = true;
             Dispose(true);
         }
 
+        public void StopWorkerProcessAsync()
+        {
+            bool capabilityEnabled = !string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.HandlesWorkerTerminateMessage));
+            if (capabilityEnabled)
+            {
+                var gracePeriod = TimeSpan.FromMilliseconds((double)WorkerConstants.ProcessExitTimeoutInMilliseconds);
+
+                var workerTerminate = new WorkerTerminate()
+                {
+                    GracePeriod = Duration.FromTimeSpan(gracePeriod)
```

Comment:
Can you remind me how will the language workers use this grace period value when they get this GRPC message?

Reply:
With grace period, we are just telling the worker that Host will wait for some time (grace period) before killing the process. It gives an ability to worker to perform needed cleanup tasks and gracefully shutdown the process within this time limit otherwise Host will kill it after timeout. Previously, Host was directly killing the process without giving any ability to worker to perform graceful shutdown.